packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -175,6 +166,51 @@ export function canonicalizeEntryConfig(
   };
 }
 
+export function canonicalizeBoundsCheckConfig(
+  boundsCheckConfig: GPCProofEntryBoundsCheckConfig
+): GPCProofEntryBoundsCheckConfig {
+  const { inRange, notInRange } = boundsCheckConfig;
+
+  // Throw if an invalid interval is specified to avoid invalid
+  // canonicalisations.
+  for (const interval of [inRange, notInRange]) {
+    if (interval && interval.min > interval.max) {
+      throw new Error(
+        `Invalid bounds check interval ${interval} in config cannot be canonicalized. The minimum value must be less than or equal to the maximum value.`
+      );
+    }
+  }
+
+  return {
+    ...(!inRange && !notInRange
+      ? {}
+      : !inRange
+      ? { notInRange: notInRange }
+      : !notInRange
+      ? { inRange: inRange }
+      : // inRange\notInRange = [inRange.min, ⍵] for some ⍵.
+      notInRange.min > inRange.min && notInRange.max >= inRange.max
+      ? {
+          inRange: {
+            min: inRange.min,
+            max: _.min([notInRange.min - 1n, inRange.max]) as bigint
+          }
+        }
+      : // inRange\notInRange = [⍺, inRange.max] for some ⍺.
+      notInRange.min <= inRange.min && notInRange.max < inRange.max
+      ? {
+          inRange: {
+            min: _.max([notInRange.max + 1n, inRange.min]) as bigint,
+            max: inRange.max
+          }
+        }
+      : {
```

Comment:
A comment explaining the remaining subcases and why they simplify down to one output would be welcome.  I think there are 3 subcases here:
- inRange within notInRange is ruled out by the check phase
- notInRange within inRange "simplifies" to 2 valid ranges, but this output is an equally-valid way to say that
- non-overlapping is the final case in which case the two are independent